src/crypto/CHIPCryptoPAL.h

```diff
@@ -1355,24 +1355,26 @@ class SymmetricKeyContext
     virtual ~SymmetricKeyContext() = default;
     /**
      * @brief Perform the message encryption as described in 4.7.2. (Security Processing of Outgoing Messages)
-     * @param[in,out] plaintext     Outgoing message payload.
+     * @param[in] plaintext     Outgoing message payload.
      * @param[in] aad           Additional data (message header contents)
      * @param[in] nonce         Nonce (Security Flags | Message Counter | Source Node ID)
-     * @param[out] out_mic      Outgoing Message Integrity Check
+     * @param[out] mic          Outgoing Message Integrity Check
+     * @param[out] ciphertext   Outgoing encrypted payload
```

Comment:
It's worth documenting that it's safe to have `plaintext` and `ciphertext` point to the same memory.

And also that ciphertext must be at least as big as plaintext when this function is called.